packages/auth-construct/src/construct.ts

```diff
@@ -1017,86 +1009,81 @@ export class AmplifyAuth
       produce: () => {
         const outputProviders = [];
         const userPoolProviders = this.resources.userPool.identityProviders;
-        if (userPoolProviders) {
-          for (const provider of userPoolProviders) {
-            const providerType =
-              provider.node['_children']['Resource']['providerType'];
-
-            if (providerType === 'Google') {
-              outputProviders.push('GOOGLE');
-            }
-            if (providerType === 'Facebook') {
-              outputProviders.push('FACEBOOK');
-            }
-            if (providerType === 'SignInWithApple') {
-              outputProviders.push('SIGN_IN_WITH_APPLE');
-            }
-            if (providerType === 'LoginWithAmazon') {
-              outputProviders.push('LOGIN_WITH_AMAZON');
-            }
-            if (providerType === 'OIDC') {
-              outputProviders.push(
-                provider.node['_children']['Resource']['providerName']
-              );
-            }
-            if (providerType === 'SAML') {
-              outputProviders.push(
-                provider.node['_children']['Resource']['providerName']
-              );
-            }
+        if (!userPoolProviders || userPoolProviders.length === 0) {
+          return;
+        }
+        for (const provider of userPoolProviders) {
+          const providerResource = provider.node.findChild(
+            'Resource'
+          ) as CfnUserPoolIdentityProvider;
```

Comment:
Do we need any validations here that the object has the expected shape?

Reply:
updated